ext/src/ruby_api/linker.rs

```diff
@@ -54,8 +54,10 @@ impl Linker {
 
         let mut inner: LinkerImpl<StoreData> = LinkerImpl::new(engine.get());
         if wasi {
-            wasi_common::sync::add_to_linker(&mut inner, |s| s.wasi_ctx_mut())
-                .map_err(|e| error!("{}", e))?
+            wasmtime_wasi::preview0::add_to_linker_sync(&mut inner, |s| s.wasi_ctx_mut())
```

Comment:
The old implementation added both preview 0 and preview 1 to the linker so copied that here. There's an argument to be made this should only add preview 1 and not preview 0.